sdks/js/DittoChatCore/src/useChat.ts

```diff
@@ -0,0 +1,52 @@
+import { Ditto } from "@dittolive/ditto";
+import { useStore } from "zustand";
+import { createStore, StoreApi } from "zustand/vanilla";
+import { useShallow } from "zustand/react/shallow";
+import { createRoomSlice, RoomSlice } from "./slices/useRooms";
+import { useMemo } from "react";
+import { ChatUserSlice, createChatUserSlice } from "./slices/useChatUser";
+import { MessageSlice, createMessageSlice } from "./slices/useMessages";
+import Room from "./types/Room";
+
+export type DittoConfParams = {
+  ditto: Ditto | null;
+  userId: string;
+  userCollectionKey: string;
+};
+
+export type CreateSlice<T> = (set: any, get: any, params: DittoConfParams) => T;
+
+export type ChatStore = RoomSlice & ChatUserSlice & MessageSlice;
+
+export let chatStore: StoreApi<ChatStore> | null = null;
```

Comment:
`export let` variables are a big anti-pattern. What was the reasoning behind this?